examples/knative/python/kn-py-vm-attr/handler.py

```diff
@@ -0,0 +1,98 @@
+from flask import Flask, request, jsonify
+from cloudevents.http import from_http
+import logging, json
+from vcenter import Session
+from datetime import date
+
+logging.basicConfig(level=logging.DEBUG,format='%(asctime)s %(levelname)s %(name)s %(threadName)s : %(message)s')
+
+app = Flask(__name__)
+@app.route('/', methods=['GET','POST'])
+def echo():
+    if request.method == 'GET':
+        sc = 200
+        msg = 'POST to this endpoint to apply custom attributes to a VM object within a cloudEvent message'
+        message = {
+            'status': sc,
+            'message': msg,
+        }
+        resp = jsonify(message)
+        resp.status_code = sc
+        return resp
+
+    if request.method == 'POST':
+        try:
+            event = from_http(request.headers, request.get_data(),None)
+
+            data = event.data
+            # hack to handle non JSON payload, e.g. xml
+            if not isinstance(data,dict):
+                data = str(event.data)
+
+            cevent = {
+                "attributes": event._attributes,
+                "data": data
+            }
+            app.logger.debug(f'"***cloud event*** {json.dumps(cevent)}')
+
+            # CloudEvent - simple validation
+            ref_vm = cevent['data']['Vm']['Vm']
+            ref_user = cevent['data']['UserName']
+            subject = cevent['attributes']['subject']
+
+            vc_s = Session()
```

Comment:
nit: a future optimization might want to move session initialization outside the handler to improve throughput, avoid vc auth throttling, etc.

Reply:
Mention added to README/Limitations